### PR TITLE
Discover frametype dynamically if passed a cache file

### PR DESCRIPTION
**Pull request by tjma12**
In the current code, when a cache file is passed to `get_timeseries_dict`, it assigns the frametype of the first channel to all channels listed in the config (in the `frametypes` dict on `L469`) and attempts to read all channels from this frametype. This causes failures when a cache file contains frames of more than one type. After this change, users can define frametypes for several channels in a config file and they will all be read from frames correctly. If frametypes are not provided in the config, the standard process for guessing a frametype will commence.

Tested this with a cache file that contains several frametypes: https://ldas-jobs.ligo.caltech.edu/~thomas.massinger/detchar/nonsens-review/summary/C01/day/20190418/compare_strain/

And tested on a config that reads directly from frames without a cache file: https://ldas-jobs.ligo.caltech.edu/~thomas.massinger/summary/h1-lomon-frame-test/1253577618-1253580618/isc/asc_local_oscillators/

### CODE REVIEW

**Comment by tjma12** [comment]
@alurban Thanks for catching that corner case! It looks like CI is now satisfied

File: gwsumm/data/timeseries.py [hunk 1]
```diff
@@ -475,7 +475,8 @@ def get_timeseries_dict(channels, segments, config=GWSummConfigParser(),
             for channel in allchannels:
                 channel = get_channel(channel)
                 ifo = channel.ifo
-                ftype = find_frame_type(channel)
+                ftype = (None if cache and not channel.frametype
+                        else find_frame_type(channel))
```
Comment (alurban) [inline gwsumm/data/timeseries.py]: @tjma12, with profuse apologies for how nitpicky this is, can you add an extra leading whitespace here for PEP-8 compliance?

```suggestion
                         else find_frame_type(channel))
```
